doc/flash-manipulation.md

```diff
@@ -0,0 +1,136 @@
+- Buffer the intended zero page (first 64 bytes)
+- Receive the program via I2C and write it to flash, except the first page
+- Once all pages except the first have been written, activate the watchdog
+- Infinite loop
+- The watchdog will reset the SMC and jump to post_reset
+- post_reset loads the new zp from ram and writes it to flash
+- Jump to main (the new program)
+- The new program starts as if it is just powered on, causing power supply to be cut
+   - In this state, it is safe to disconnect power supply, but it is not needed, as the power button can be used
+
+v2, bad version:
+- Install a custom zp used for USI interrupts AND a new reset vector to jump to "post_reset"
+- Buffer the intended zero page (first 64 bytes)
+- Receive the program via I2C and write it to flash, except the first page
+- Once all pages except the first have been written, start infinite loop WITHOUT ACTIVATING WATCHDOG (this is the problem with this version)
+   - If power supply is disconnected here, you will get a bricked SMC
+   - If SMC is restarted using the reset pin, SMC will not be bricked
+- Once SMC restarts (due to reset pin or disconnected power), it jumps to "post_reset"
+- post_reset loads the new zp from ram and writes it to flash
+   - If reset pin was used, ram contents and zp is preserved
+   - If power supply was removed, ram contains random data, and will in all likelyhood brick the SMC
+- Jump to main (the new program)
+- The new program starts as if it is just powered on
+
+v3:
+- The bootloader main function is always called after SMC reset. The main function checks if the Reset button is pressed. If not, the bootloader
+  jumps to the firmware main entry point. If the Reset button is pressed, the computer is turned on, and the
+  firmware update process is started.
+- The bootloader communicates via I2C using the USI hardware module, but without interrupts. There is no need for a custom zp during the update anymore.
+- On receiving the first page (first 64 bytes) the bootloader deletes the whole firmware section of the flash memory.
+- Each received page is then written to flash memory.
+- After all pages have been written to flash memory, the SMC is reset by the watchdog timer. After reset, execution starts again from the
+  bootloader main function. Unless the Reset button is pressed at that time, the firmware is started normally. The computer can be turned on
+  by pressing the Power button.
+
+Changelog:
+- Bundle 1: Proof of concept, for use with a SMC without button protection
+- Bundle 2: New bootloader programmer for SMC with button protection, also more user friendly
+- Bundle 3: New bootloader programmer, changing text as button combo was changed from pwr+NMI to pwr+reset
+- Bundle 4: Improve readme.txt
+- Bundle 5: Specifies that SMC version 47.2.0 is minimum required version
+- Bundle 6: Markdown formatting, bootloader 3 support, some simplifactions
```

Comment:
The thing called "Bundle 6" is not part of an isolated bundle anymore. "Version 6" (of this document) may be a more descriptive name?